flink-java/src/main/java/org/apache/flink/api/java/operators/SortPartitionOperator.java

```diff
@@ -59,6 +62,14 @@ public SortPartitionOperator(DataSet<T> dataSet, String sortField, Order sortOrd
 		this.appendSorting(flatOrderKeys, sortOrder);
 	}
 
+	public <K> SortPartitionOperator(DataSet<T> dataSet, Keys.SelectorFunctionKeys<T, K> sortKey, Order sortOrder, String sortLocationName) {
+		super(dataSet, dataSet.getType());
+		this.sortLocationName = sortLocationName;
+
+		int[] flatOrderKeys = getFlatFields(sortKey);
```

Comment:
This does not work. 
`SelectorFunctionKeys` need to be handled completely different. In short, you need to inject a MapFunction<IN, Tuple2<KEY, IN>> that extracts the key and emits a Tuple2 with the extracted key and the original input record.
The data set is then sorted on the first tuple field (the key field) and the records are unwrapped after the sort by another `MapFunction<Tuple2<KEY, IN>, IN>`. Have a look at how the PartitionOperator handles KeySelectors.